Assets/Scripts/BlockBehaviour.cs

```diff
@@ -107,4 +108,22 @@ private void TranslateBlock()
             _isTranslating = false;
         }
     }
+
+    private void OnCollisionEnter(Collision other)
+    {
+        if (other.gameObject.name == "Character")
+        {
+            _isPlayerTouched = true;

```

Comment:
Once the player has touched the block, there is no way to figure out when the player has retouched it. I think it would be better to use delegate to inform the plugins when a player touches the block. Can look at OnFaceClick for example.